internal/pkg/process/info.go

```diff
@@ -152,3 +170,38 @@ func (i *Info) GetFunctionReturns(name string) ([]uint64, error) {
 
 	return nil, fmt.Errorf("could not find returns for function %s", name)
 }
+
+// onceResult is an object that will perform exactly one action if that action
+// does not error. For errors, no state is stored and subsequent attempts will
+// be tried.
+type onceResult[T any] struct {
```

Comment:
I would not make it as a generic if we need it only for allocations.
It is a lessons learned from C# that when generics where added then everyone was adding it to places like here where it was not necessary.
With making it for allocations only we already show where it is needed + makes the code a little simpler for those not familiar with generics.

This could also help us with coming up with a simpler name. E.g. `eventualResult[Allocation]` is more complex than `eventualAlloc`.

Reply:
This can be flattened into the type if we don't need it to be generic. Updating.

Reply:
Updated. PTAL.